### PR TITLE
Add tests for `CmaEsSampler`.

### PR DESCRIPTION
**Pull request by c-bata**
## Motivation

Follow-up #4184.

*Let me this PR ready for review after merged #4184.*

## Description of the changes

Add test cases to verify that `CmaEsSampler` restores the correct optimizer and updates the distribution with the correct solution trials.

### CODE REVIEW

**Comment by c-bata** [comment]
@gen740 @contramundum53 Could you review this PR if you have time?

File: tests/samplers_tests/test_cmaes.py [hunk 1]
```diff
@@ -400,28 +402,147 @@ def test_restore_optimizer_keeps_backward_compatibility() -> None:
...
+    assert optimizer is None
+    assert n_restarts == 0
 
-    system_attrs: Dict[str, Any] = sampler._split_optimizer_str(optimizer_str)
-    assert len(system_attrs) > 1
-    system_attrs["cma:n_restarts"] = 1
+    def objective(trial: optuna.Trial) -> float:
+        x1 = trial.suggest_float("x1", -10, 10, step=1)
+        x2 = trial.suggest_float("x2", -10, 10)
+        return x1**2 + x2**2
```
Comment (gen740) [inline tests/samplers_tests/test_cmaes.py:416]: This objective function appears so many times, how about defining this function on the top of this test?

**Review by contramundum53 — APPROVED** [review]
LGTM!

File: tests/samplers_tests/test_cmaes.py (hunk 1 above)
Comment (c-bata) [reply tests/samplers_tests/test_cmaes.py:416]: @gen740 Thank you for your review!

I understand your motivation. Obviously, this is not optimal in DRY mind. It's a matter of preference, but reusing the same objective function in multiple test cases often made tests complex. (e.g. `prepare_study_with_trials` function, which is used in visualization tests, contains a lot of implicit specifications and lots of unit tests depends on it. See https://github.com/optuna/optuna/pulls?q=is%3Apr+author%3Ac-bata+prepare_study_with_trials for details.)

So if it's a not your strong opinion, I suggest keeping this function not DRY. What do you think?

Comment (gen740) [reply tests/samplers_tests/test_cmaes.py:416]: I see. Indeed, reusing an objective function is sometimes hard to read...
I think that's fine to define for each test.

**Review by gen740 — APPROVED** [review]
LGTM!

**Comment by c-bata** [comment]
@contramundum53 Please merge this PR since this PR has two approvals.

**Pull request merged**